README.md

```diff
@@ -112,6 +113,109 @@ plugins: [
 ];
 ```
 
+## Running Babel on the generated code
+
+By setting `options.transformGenerated: true`, you can configure rollup-plugin-babel to run Babel on the output files instead of the input files. This can be used to perform code transformations on the resulting chunks and is the only way to transform Rollup's auto-generated code. By default, the plugin will be applied to all outputs:
+
+```js
+// rollup.config.js
+import babel from 'rollup-plugin-babel';
+export default {
+	input: 'main.js',
+	plugins: [babel({ transformGenerated: true })],
+	output: [
+		{ file: 'bundle.cjs.js', format: 'cjs' },
+		{ file: 'bundle.esm.js', format: 'esm' },
+	],
+};
+```
+
+If you only want to apply it to specific outputs, you can use it as an output plugin:
+
+```js
+// rollup.config.js
+import babel from 'rollup-plugin-babel';
+export default {
+	input: 'main.js',
+	output: [
+		{ file: 'bundle.js', format: 'esm' },
+		{
+			file: 'bundle.min.js',
+			format: 'esm',
+			plugins: [babel({ transformGenerated: true })],
+		},
+	],
+};
+```
+
+The `include`, `exclude` and `extensions` options are ignored when the `transformGenerated` option is used and will produce warnings, and there are a few more points to note that users should be aware of.
+
+### Finding .babelrc files
+
+Depending on what options are used, this plugin will start looking for `.babelrc` files starting in the directory of the file indicated by `output.file` or in the directory indicated by `output.dir`. If neither of these options is used, e.g. when rendering to stdout or when using the JavaScript API, then `.babelrc` files are ignored and options need to be passed in manually.
+
+### Injected helpers
+
+By default, helpers e.g. when transpiling classes will be inserted at the top of each chunk. In contrast to when applying this plugin on the input files, helpers will not be deduplicated across chunks.
+
+You can however use external helpers by adding the `@babel/external-helpers` plugin, which will look for helpers on a global variable named `babelHelpers`:
+
+```js
+rollup.rollup({...})
+.then(bundle => bundle.generate({
+  format: 'esm',
+  plugins: [babel({
+    transformGenerated: true,
+    presets: ['@babel/env'],
+    plugins: ['@babel/external-helpers']
+  })]
+}))
+```
+
+```js
+// input
+export default class Foo {}
+
+// output
+var Foo = function Foo() {
+	babelHelpers.classCallCheck(this, Foo);
+};
+
+export default Foo;
+```
+
+Alternatively, you can use imported runtime helpers by adding the `@babel/transform-runtime` plugin. Note that to produce valid code e.g. with `@babel/env`, it is very recommended to configure the format of modules for that plugin to match the output format used by Rollup:
+
+```js
+rollup.rollup({...})
+.then(bundle => bundle.generate({
+  format: 'cjs',
+  plugins: [babel({
+    transformGenerated: true,
+    presets: [['@babel/env', { modules: 'cjs' }]],
```

Comment:
I dont believe this is needed - babel shouldnt apply cjs transform when used in combination with Rollup (in such combination this is Rollup’s job)

I would recommend using `useESModules` flag appropriately for transform-runtime in here.

Reply:
When I change it the way you recommend, i.e.
```
presets: [[@babel/env'],
plugins: [['@babel/transform-runtime', { useESModules: false }]],
```
in the test `allows transform-runtime to be used instead of bundled helpers for CJS output`, then I get

```js
'use strict';

import _classCallCheck from "@babel/runtime/helpers/classCallCheck";

var Foo = function Foo() {
  _classCallCheck(this, Foo);
};

module.exports = Foo;
```

which looks VERY broken to me.

The problem is, we are running Babel AFTER Rollup has done its job—there will be no more transformations by Rollup at this point. Thus
- The Babel plugin has complete freedom what it does with the output, but
- It needs to preserve import and export semantics

But maybe I should update the Readme to highlight this better: This is not an issue specific to the runtime transform but to any Babel plugin that modifies imports or exports.

Reply:
> It needs to preserve import and export semantics

That's very true, but also those imports should be inserted with that in mind by Babel. So I'm wondering why they were not. I wonder that maybe this is caused by this:
https://github.com/rollup/rollup-plugin-babel/blob/6a99fdf957308f77c320165eb50e0af6b147f081/src/index.js#L25 
and maybe that should be changed based on the generated format in this scenario? This is just a wild guess - it needs more investigation (if you don't have time do to it, I can get to it in a few days, just let me know)

Reply:
That should definitely be investigated before proceeding with this. I will definitely not find time to look into this today, maybe end of the week.

Reply:
I've added a commit fixing this - https://github.com/rollup/rollup-plugin-babel/pull/333/commits/8e38908f6bb1c9332ff61b09b65092337744e25c . Please take a loot if this makes sense. cc @loganfsmyth - I would appreciate your review on this one as well.

Reply:
The meaning of `supportsStaticESM` is `this tool knows how to parse ESM`, not "this file is ESM". The issue here is, I'd assume, more related to [`sourceType`](https://babeljs.io/docs/en/options#sourcetype) defaulting to `module`. We may want to consider defaulting `sourceType` to `script` when `format !== "esm`?

Reply:
I agree that makes sense. Will play around with this.

Reply:
Using `sourceType` provides much better output for the transform-runtime test and also avoids the unnecessary "use strict" in the other test. I am undecided `supportsStaticESM`, though. It does not seem to have adverse effects when we use Babel to transpile ESM to another format, but not adding it will cause some plugins to automatically go for CJS imports, which seems weird to me, especially if you wanted to keep the ESM format. My gut feeling goes towards keeping it if the output format is ESM.

Reply:
With the definition given by @loganfsmyth :
> The meaning of supportsStaticESM is this tool knows how to parse ESM, not "this file is ESM".

when we use babel as output plugin then it seems that the correct thing to state is "this tool doesn't know how to parse ESM" as there won't be any more parsing on the tool's side. It's slightly unsettling to state that when we know that the target format is ESM though

🤷‍♂

Reply:
I should rephrase that if my reading implied that. I'd think of it as meaning that the output of Babel will be consumed by a tool that knowns about esm. When compiling a bundle, it seems like setting it true when when the format is esm makes sense. Otherwise if you set the format the esm in rollup but then tried to run preset-env, it would try to compile the file to cjs by default.